Added documentation (#13)
exasol/sphinx-github-pages-generator: doc/user_guide/user_guide.rst

===========================
Building the Documentation
===========================

We are using Sphinx for generating the documentation,
because it is the default documentation tool for Python projects.
Sphinx supports API documentation generation for Python and with plugins also for other languages.
Furthermore, it supports reStructuredText with proper cross-document references.

---------------------------
Needed Repository structure
---------------------------

This Project uses Sphinx for the generation of the documentation. Therefore, all files you want to use as a
source for your documenation have to be compatible with Spinx. You can find Shpinx's
documentation `here <https://www.sphinx-doc.org/en/master/>`_.

In general:

You will need a base directory for your documentation source files. We use "repository_root/doc",
but you can set it to another folder.
Inside the documentation root directory, you need at least a minimal "conf.py" to configure
Sphinx and a "index.rst"
for Sphinx to use as a root for your documentation. You can use :code:`sphinx-quickstart` to generate stubs for these.

> Inside the index.rst you can linc to other parts of your documentation using the :code:`toctree` directive.

Reviewer: ```suggestion
Inside the index.rst you can link to other parts of your documentation using the :code:`toctree` directive.
```

Reviewer: Is it really `toctree` or should it be `doctree`?

You can also include plain text, or documentation for specific functions or objects by using the
:code:`autosummary`,
:code:`automodule`,
:code:`autoclass` and
:code:`autoexception` directives, which will import the docstring for the given objects automatically.

> Here is the code from te index.rst of this project, which generates the documentation if the api. To see the

Reviewer: "Here" needs a link, right?

generated pages see :ref:`api_ref_target`.

.. code:: rst

    .. autosummary::
       :toctree: api

       exasol_sphinx_github_pages_generator

-----------------------------
Call the Sphinx generator
-----------------------------

Once Sphinx generator is installed in your environment, you can import and use it in a
python script like this:

.. code:: py

    import exasol_sphinx_github_pages_generator.deploy_github_pages as deploy_github_pages
    import sys

    if __name__ == "__main__":
        deploy_github_pages.deploy_github_pages(sys.argv[1:])

> Then call the Script using command line parameters like this:

Reviewer: ```suggestion
Then call the script using command line parameters like this:
```

                                             "--push_origin", "origin",
                                             "--push_enabled", "push",
                                             "--source_branch", "source_branch",
                                             "--source_dir", cwd,
                                             "--module_path", "../package/module-path1", "../package/module-path2"])

The generator has to be called from the working directory containing the index.rst file.

-------
Options
-------

Calling the module with "-h" will print the help page for the generator.

.. code:: py

    deploy_github_pages.deploy_github_pages(["-h"])

Parameters:

.. autoclass:: exasol_sphinx_github_pages_generator.deploy_github_pages.GithubPagesDeployer

======================================================
Building the Documentation interactively during coding
======================================================

# todo should we add this again? otherwise people can just use the sphinx commands

======================================================
Building the Documentation in the CI
======================================================

> The documentation for this project is build in GitHub Actions using this project. We will use the process as an example here.

Reviewer: ```suggestion
The documentation for this project is built on GitHub Actions using this project. We will use the process as an example here.
```
